pool/object_test.go

```diff
@@ -0,0 +1,555 @@
+	case objID != x.objID:
+		return nil, errors.New("[test] wrong object ID")
+	case !assert.ObjectsAreEqual(signer, x.sgnr):
+		return nil, errors.New("[test] wrong signer")
+	case !assert.ObjectsAreEqual(opts, x.opts):
+		return nil, errors.New("[test] wrong options")
+	}
+	return x.hs, x.err
+}
+
+type objectHashOnlyClientWrapper struct {
+	mockedClientWrapper
+	c objectHashOnlyClient
+}
+
+func (x objectHashOnlyClientWrapper) getClient() (sdkClientInterface, error) { return x.c, nil }
+
+func TestPool_ObjectHash(t *testing.T) {
+	ctx := context.Background()
+	cnrID := cidtest.ID()
+	objID := oidtest.ID()
+	usr := usertest.User()
+
+	var hashOpts client.PrmObjectHash
+	hashOpts.WithinSession(sessiontest.Object())
+	hashOpts.WithBearerToken(bearertest.Token())
+	hashOpts.MarkLocal()
+	hashOpts.WithXHeaders("k1", "v1", "k2", "v2")
+	hashOpts.TillichZemorAlgo()
+	hashOpts.SetRangeList(1, 2, 3, 4)
+	hashOpts.UseSalt([]byte("any_salt"))
+
+	hashClient := objectHashOnlyClient{
+		cnr:   cnrID,
+		objID: objID,
+		sgnr:  usr,
+		opts:  hashOpts,
+		hs:    [][]byte{[]byte("hash1"), []byte("hash2")},
+		err:   errors.New("any error"),
+	}
+	endpoints := []string{"localhost:8080", "localhost:8081"}
+	nodes := make([]NodeParam, len(endpoints))
+	cws := make([]objectHashOnlyClientWrapper, len(endpoints))
+	for i := range endpoints {
+		nodes[i].address = endpoints[i]
+		cws[i].addr = endpoints[i]
+		cws[i].c = hashClient
+	}
+
+	var poolOpts InitParameters
+	poolOpts.setClientBuilder(func(endpoint string) (internalClient, error) {
+		ind := slices.Index(endpoints, endpoint)
+		if ind < 0 {
+			return nil, fmt.Errorf("unexpected endpoint %q", endpoint)
+		}
+		return &cws[ind], nil
+	})
+	p, err := New(nodes, usertest.User().RFC6979, poolOpts)
+	require.NoError(t, err)
+	require.NoError(t, p.Dial(ctx))
+	t.Cleanup(p.Close)
+
+	hs, err := p.ObjectHash(context.Background(), cnrID, objID, usr, hashOpts)
+	require.Equal(t, err, hashClient.err)
+	require.Equal(t, hs, hashClient.hs)
+}
+
+type objectSearchOnlyClient struct {
+	noOtherClientCalls
+	// expected input
+	cnr  cid.ID
+	sgnr user.Signer
+	opts client.PrmObjectSearch
+	// ret
+	rdr *client.ObjectListReader
+	err error
+}
+
+func (x objectSearchOnlyClient) ObjectSearchInit(ctx context.Context, cnr cid.ID, signer user.Signer, opts client.PrmObjectSearch) (*client.ObjectListReader, error) {
+	switch {
+	case ctx == nil:
+		return nil, errors.New("[test] nil context")
+	case cnr != x.cnr:
+		return nil, errors.New("[test] wrong container")
+	case !assert.ObjectsAreEqual(signer, x.sgnr):
+		return nil, errors.New("[test] wrong signer")
+	case !assert.ObjectsAreEqual(opts, x.opts):
+		return nil, errors.New("[test] wrong options")
+	}
+	return x.rdr, x.err
+}
+
+type objectSearchOnlyClientWrapper struct {
+	mockedClientWrapper
+	c objectSearchOnlyClient
+}
+
+func (x objectSearchOnlyClientWrapper) getClient() (sdkClientInterface, error) { return x.c, nil }
+
+func TestPool_ObjectSearchInit(t *testing.T) {
```

Comment:
Why duplicating prologue in these? Likely we can have less code and still separate subtests for different operations.

Reply:
prefer independent tests for methods